### PR TITLE
OPSEXP-2747 Avoid configuring repository with search enterprise for community

### LINKED ISSUE #907: Default inventory requires change to work with community edition
#### Bug description

After running the ansible-playbook with arguments for local deployment (pipenv run ansible-playbook playbooks/acs.yml -i inventory_local.yml -e "@community-extra-vars.yml"), I tried accessing the service via HTTPS and it died few seconds after trying to login (which it complained that it couldn't authenticate). 
Digging through the alfresco logs, I only found a mention of missing "elasticsearch" beans from Tomcat, and typical Java errors.
After many hours of looking at documentation and extensive search, I looked at our test instance in Docker, and I noticed when comparing the two that ansible didn't deploy Solr search engine. 
After looking at inventory_local.yml, I tried adding repository: variable and re-ran it. 
Afterwards, the Alfresco service ran without any noticeable problem.

So the part in inventory_local.yml looked like this in the end:
```
    # Solr search engine (alternative to Enterprise Search)
    search:
      children:
        repository:
```

#### Target OS

AlmaLinux 9 (tried it here first, but changed to RockyLinux since AL is not officially supported)
RockyLinux 9 (final deployment)

#### Host OS

RockyLinux 9

#### Playbook version

v2.9.0

#### Ansible error

No error, deployment was successful. The Solr parts were marked as [Skipped]

##### ansible --version

```shell
ansible [core 2.15.11]
  config file = /root/alfresco-ansible-deployment/ansible.cfg
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /root/.local/share/virtualenvs/alfresco-ansible-deployment-qhTOnLa5/lib/python3.9/site-packages/ansible
  ansible collection location = /root/.ansible/collections:/usr/share/ansible/collections
  executable location = /root/.local/share/virtualenvs/alfresco-ansible-deployment-qhTOnLa5/bin/ansible
  python version = 3.9.18 (main, May 16 2024, 00:00:00) [GCC 11.4.1 20231218 (Red Hat 11.4.1-3)] (/root/.local/share/virtualenvs/alfresco-ansible-deployment-qhTOnLa5/bin/python)
  jinja version = 3.1.4
  libyaml = True

```

##### ansible-config dump --only-changed

```shell
ANSIBLE_PIPELINING(/root/alfresco-ansible-deployment/ansible.cfg) = True
CONFIG_FILE() = /root/alfresco-ansible-deployment/ansible.cfg
```

##### ansible-inventory -i your_inventory_file --graph

```shell
@all:
  |--@ungrouped:
  |--@repository:
  |  |--localhost
  |--@database:
  |  |--@repository:
  |  |  |--localhost
  |--@activemq:
  |  |--@repository:
  |  |  |--localhost
  |--@search:
  |  |--@repository:
  |  |  |--localhost
  |--@search_enterprise:
  |  |--@repository:
  |  |  |--localhost
  |--@elasticsearch:
  |  |--@repository:
  |  |  |--localhost
  |--@identity:
  |--@nginx:
  |  |--@repository:
  |  |  |--localhost
  |--@acc:
  |  |--@repository:
  |  |  |--localhost
  |--@adw:
  |  |--@repository:
  |  |  |--localhost
  |--@transformers:
  |  |--@repository:
  |  |  |--localhost
  |--@syncservice:
  |  |--@repository:
  |  |  |--localhost
  |--@other_repo_clients:
  |--@trusted_resource_consumers:
  |  |--@repository:
  |  |  |--localhost
  |  |--@nginx:
  |  |  |--@repository:
  |  |  |  |--localhost
  |  |--@adw:
  |  |  |--@repository:
  |  |  |  |--localhost
  |  |--@other_repo_clients:
  |--@external_activemq:
  |--@external_elasticsearch:
  |--@external_identity:
  |--@external:
  |  |--@external_activemq:
  |  |--@external_elasticsearch:
  |  |--@external_identity:
  |  |--@other_repo_clients:
```

##### pip list

```shell
Package             Version
------------------- -------
ansible-core        2.15.11
cffi                1.16.0
cryptography        42.0.5
importlib-resources 5.0.7
Jinja2              3.1.4
jmespath            1.0.1
MarkupSafe          2.1.5
packaging           24.0
pip                 24.1
pycparser           2.22
PyYAML              6.0.1
resolvelib          1.0.1
setuptools          70.1.0
wheel               0.43.0
```

### PR DESCRIPTION
**Pull request by gionn**
ref: OPSEXP-2747

Fix #907

### CODE REVIEW

**Review by alxgomz — CHANGES_REQUESTED** [review]
Please add a pre-check which fails the play in case inventory is configured with community edition awith search enterprise hosts and no search hosts and always skip searc enterprise play when community is set

**Review by pmacius — APPROVED** [review]

File: inventory_local.yml [hunk 1]
```diff
@@ -16,11 +16,12 @@ all:
       children:
         repository:
 
-    # Solr search engine (alternative to Enterprise Search)
+    # Solr search engine (alternative to Enterprise Search - only option for Community)
```
Comment (gionn) [inline inventory_local.yml:19]:
```suggestion
    # Solr search engine (an alternative to Enterprise Search, and the only available option for the Community edition)
```

File: inventory_local.yml [hunk 2]
```diff
@@ -16,11 +16,12 @@ all:
...
     search:
       children:
+        # repository:
 
-    # Enterprise Search (default search engine)
+    # Enterprise Search (default search engine - not available for Community)
```
Comment (gionn) [inline inventory_local.yml:24]:
```suggestion
    # Enterprise Search (default search engine - not available in Community edition)
```

File: inventory_ssh.yml [hunk 3]
```diff
@@ -27,11 +27,13 @@ all:
         activemq_1:
           ansible_host: targetIP
 
-    # Solr search engine (alternative to Enterprise Search)
+    # Solr search engine (alternative to Enterprise Search - only option for Community)
```
Comment (gionn) [inline inventory_ssh.yml:30]:
```suggestion
    # Solr search engine (an alternative to Enterprise Search, and the only available option for the Community edition)
```

File: inventory_ssh.yml [hunk 4]
```diff
@@ -27,11 +27,13 @@ all:
...
     search:
       hosts:
+        # search_1:
+        #   ansible_host: targetIP
 
-    # Enterprise Search (default search engine)
+    # Enterprise Search (default search engine - not available for Community)
```
Comment (gionn) [inline inventory_ssh.yml:36]:
```suggestion
    # Enterprise Search (default search engine - not available in Community edition)
```

**Review by alxgomz — APPROVED** [review]

**Pull request merged**